lib/jobs/message.job.ts

```diff
@@ -0,0 +1,142 @@
+import * as i18nService           from '../services/i18n'
+import { MessagesApi }            from 'sunshine-conversations-client'
+import { createJob }              from './job'
+import { unsafe }                 from '../utils/errors'
+import { Json }                   from '../typings/lang'
+import config                     from '../config'
+import db                         from "../db"
+import _                          from 'lodash'
+import {
+  AuthorType,
+  Conversation,
+  ConversationType,
+  DeliveryStatus,
+  Message
+} from "@prisma/client"
+
+const i18n = i18nService.initialize();
+
+// ---------------------------
+// ~ TYPES
+// ---------------------------
+
+type MessageAction = (
+  //
+  // Add any supported actions of this job here
+  //
+  "deliver"
+)
+
+// ---------------------------
+// ~ ACTION METHODS
+// ---------------------------
+
+/**
+ * Returns true if the message can be sent to sunshine
+ *
+ * @param {Conversation} conversation
+ * @param {Message} message
+ * @returns
+ */
+const canDeliverMessage = (conversation: Conversation, message: Message) => {
+  return (
+    conversation.type === ConversationType.CUSTOMER &&
+    conversation.sunshineConversationId &&
+    !message.sunshineMessageId &&
+    message.authorType !== AuthorType.CUSTOMER && (
```

Comment:
Is it even possible for customer author type to reach here? Since our GraphQL endpoints only allow staff.

Reply:
It's not, but it doesn't matter. Writing a function with the assumption of where/when it'll be called is a ticking time bomb, better to be safer and write the function as it should be